addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1421,6 +2146,34 @@ var SnippetsMenu = Widget.extend({
             },
         });
 
+        // Adding a helper grid in all snippets in grid mode.
+        this.options.wysiwyg.odooEditor.observerUnactive('addHelperGrid');
+        const $gridModeRows = this.getEditableArea().find('.row.o_grid_mode');
+        for (let rowEl of $gridModeRows) {
+            const rowCount = parseInt(rowEl.style.getPropertyValue('--row-count'));
+            const rowGap = rowEl.style.getPropertyValue('--row-gap');
+            const columnGap = rowEl.style.getPropertyValue('--column-gap');
+
+            const helperGrid = document.createElement('div');
+            helperGrid.classList.add('o_helper_grid', 'd-none');
+
+            helperGrid.style.gridArea = `1 / 1 / ${rowCount + 1} / -1`;
+            helperGrid.style.setProperty('--row-gap', rowGap);
+            helperGrid.style.setProperty('--column-gap', columnGap);
+
+            for (let i = 1; i <= rowCount; i++) {
+                for (let j = 1; j <= 12; j++) {
+                    const gridCell = document.createElement('div');
+                    gridCell.classList.add('o_helper_cell');
+                    gridCell.style.gridArea = `${i} / ${j} / ${i + 1} / ${j + 1}`;
+
+                    helperGrid.append(gridCell);
+                }
+            }
+            rowEl.append(helperGrid);
+        }
+        this.options.wysiwyg.odooEditor.observerActive('addHelperGrid');
```

Comment:
probably something not to be done (with bg image svg or multiple divs): I'd prefer the grid recreated each time we start using it (resize, move, etc). That should not be perf-heavy (just like other drag & drop etc)